### DISCUSSION TITLE
iptables-persistent and Homer problem ?

### DISCUSSION THREAD

**Discussion by sraimondeau** [sipcapture/homer#629] [General]
Hello,

I don't know if it is the right place to post an issue I have. anyway I need help so... :-)

I have setup a fresh Homer 7 install on a Debian 10 (test also with Debian 11) using ./homer_installer.sh bash script.
VM is hosted at OVHcloud.
My Oracle SBCs are sending HEP and everything is OK. I can see the SIP exchanges in the web interface.
I see also the increment in the psql table hep_proto_1_call (https://github.com/sipcapture/homer/wiki/Troubleshooting-Homer-7.7)

Now, I want to secure a little bit the installation.
So I install iptables-persistent with apt-get install iptables-persistent .
No rules in it, only the package.
Just after install, the count in hep_proto_1_call is incrementing correctly.
Now I reboot the server, 
After the reboot, the count in hep_proto_1 stops incrementing and so the web interface does not get the new SIP communications.
Of course, with a simple reboot without package iptables-persistent, everything is fine.
apt-get remove iptables-persistent does not solve the issue.

Have you got an idea where the problem is ?

Kind regards, Stéphane.

**Comment by lmangani** [comment]
Hello @sraimondeau 

You will need to set up explicit rules for any port you want to use and save them as persistent.

**Comment by sraimondeau** [comment]
Hello @lmangani , thank you for your reply.

This is the content of /etc/iptables/rules.v4 generated during iptables-persistent installation:

-----
# Generated by xtables-save v1.8.2 on Mon Feb 26 15:39:17 2024
*filter
:INPUT ACCEPT [0:0]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
COMMIT
# Completed on Mon Feb 26 15:39:17 2024
-----

I think my problem is not linked to the iptables rules because there are none.

**Comment by sraimondeau** [comment]
Thanks to @ewadii (https://github.com/sipcapture/homer-installer/issues/88), problem is resolved with the modification of /lib/systemd/system/heplify-server.service :
Uncomment:
#Only if you have postgresql on the same host!
#After=network.target postgresql.service

**Comment by adubovikov** [comment]
Hi [@sraimondeau](https://github.com/sraimondeau),

Good find, and thanks for posting the fix.

### Root cause (not iptables rules)

You were right: empty `/etc/iptables/rules.v4` means this was **not** a firewall block.

What happened is a **systemd startup ordering** issue on Homer 7:

- `heplify-server` started before PostgreSQL was ready (or before networking settled after boot)
- ingest looked fine until first reboot
- installing `iptables-persistent` changed boot timing enough to expose the race

So the package name is misleading here. It triggered the bug, but did not filter HEP traffic.

### Fix (Homer 7 + local PostgreSQL)

In `/lib/systemd/system/heplify-server.service` (or better: an override), enable:

```ini
[Unit]
After=network-online.target postgresql.service
Wants=network-online.target
Requires=postgresql.service
```

Then:

```bash
sudo systemctl daemon-reload
sudo systemctl restart heplify-server
```

Reference: [homer-installer#88](https://github.com/sipcapture/homer-installer/issues/88)

Use `Requires=` only if PostgreSQL is on the same host. If DB is remote, keep `After=network-online.target` and point `heplify-server` at the remote DB host.

### Quick checks when `hep_proto_1_call` stops growing

```bash
systemctl status heplify-server postgresql
journalctl -u heplify-server -b --no-pager | tail -50
ss -ulnp | grep 9060
psql -c "select count(*) from hep_proto_1_call;"
```

Typical failure signatures:

- `connection refused` / `database does not exist` at startup → DB ordering
- process running but UDP 9060 not listening → service failed early and was not restarted
- counter frozen while UDP arrives → DB write path stuck (less common after reboot)

### HEP still visible on the wire but not in DB

If `tcpdump -ni any udp port 9060` shows packets but DB count is flat, the problem is on the Homer side (service/DB), not on the Oracle SBC sender.

### Homer 11 note

Homer 11 uses `homer-core` (DuckLake/DuckDB config store). Same class of issue applies: ensure the core service starts **after** `network-online.target` and any local storage/DB dependency it needs.

For new installs, prefer **Homer 11** ([releases](https://github.com/sipcapture/homer/releases)). This thread is specific to the legacy Homer 7 + `heplify-server` + PostgreSQL stack.

Bottom line: not iptables. Fix systemd ordering for `heplify-server` vs PostgreSQL/network.
